### DISCUSSION TITLE
the different between "" and String.Empty

### DISCUSSION THREAD

**Discussion by xiaodongcool** [dotnet/runtime#68463] [Q&A] [answered]
Are there any performance and memory allocation differences between the two ?

**Comment by FaustVX** [comment]
There are a lot of answers on the internet :
https://stackoverflow.com/questions/263191/in-c-should-i-use-string-empty-or-string-empty-or-to-intitialize-a-string
https://stackoverflow.com/questions/151472/what-is-the-difference-between-string-empty-and-empty-string

**Comment by teo-tsirpanis** [comment, marked as answer]
As you can see, [both generate identical assembly](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAZgAJiAmcgYXIG8BYAKHPcouIEYAGcgMrdyAXgB85AESSA3Kw6dKfQTXFLeAOgCi+AA4YAnnJYBfIA==) (on .NET 6 at least).

**Reply by gfoidl** [reply]
Small correction: this is the [correct sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAZgAJiAmcgYXIG8BYAKHPcouIEYAGcgMrdyAXgB85AESSA3Kw6dKfQTXFLeAOgCi+AA4YAnnJYBfIA==)

See also https://github.com/dotnet/runtime/issues/68347#issuecomment-1105681840

**Reply by xiaodongcool** [reply]
the different between JIT Asm and IL ?

**Reply by teo-tsirpanis** [reply]
Oops, my mistake @gfoidl. 😳

@jyller, IL is the intermediate bytecode the C# compiler generates, and JIT Asm is the machine code the Just-in-Time compiler generates.
